frontend/src/lib/components/jobs/JobProgressBar.svelte

```diff
@@ -0,0 +1,59 @@
+<script lang="ts">
+	import { type Job } from '$lib/gen'
+	import ProgressBar from '../progressBar/ProgressBar.svelte'
+
+	export let job: Job | undefined = undefined
+	/// Progress of currently running job
+	export let scriptProgress: number | undefined = undefined;
+
+	let error: number | undefined = undefined
+	let index = 0
+	let subIndex: number = 0
+	let subLength: number  = 100
+	let length = 1
+	let nextInProgress = false
+
+	$: if (job) updateJobProgress(job);
+	$: subIndex = scriptProgress ?? 0;
+
+	function updateJobProgress(job: Job) { 
+		if (job['success'])	
+			index = 1, subLength = 0, subIndex = 0;				
+	}
+
+	let resetP: any
+
+	export function reset() {
+		resetP?.()
+		error = undefined
+		subIndex = 0
+		subLength = 100
+		length = 1
+		index = 0
+		scriptProgress = undefined
+	}
+
+</script>
+
+<style>
+  :global(.compact-progress-bar > div:nth-child(2) ) {
+  	border-radius: 0 !important;
+  	height: 12px !important;
+  }
+  :global(.compact-progress-bar > div:nth-child(1)) {
+  	display: none !important;
+  }
+</style>
```

Comment:
instead of having styles here you could add a prop to ProgressBar and conditionally apply tailwind classnames to the divs in question (twMerge can help you with that)

Reply:
Fixed